src/Symfony/Component/VarDumper/Dumper/ContextProvider/SourceContextProvider.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\VarDumper\Dumper\ContextProvider;
+
+use Symfony\Component\HttpKernel\Debug\FileLinkFormatter;
+use Symfony\Component\VarDumper\Cloner\VarCloner;
+use Symfony\Component\VarDumper\Dumper\HtmlDumper;
+use Symfony\Component\VarDumper\VarDumper;
+use Twig\Template;
+
+/**
+ * Tries to provide context from sources (class name, file, line, code excerpt, ...).
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ * @author Maxime Steinhausser <maxime.steinhausser@gmail.com>
+ */
+final class SourceContextProvider
+{
+    private $limit;
+    private $charset;
+    private $projectDir;
+    private $fileLinkFormatter;
+
+    public function __construct(int $limit = 9, string $charset = null, string $projectDir = null, FileLinkFormatter $fileLinkFormatter = null)
+    {
+        $this->charset = $charset;
+        $this->limit = $limit;
+        $this->projectDir = $projectDir;
+        $this->fileLinkFormatter = $fileLinkFormatter;
+    }
+
+    public function __invoke(): array
```

Comment:
By design, this has to be a callable as a context provider used by the server dumper. But actually, no need to use a local variable anymore since this is not targeting the 3.4 branch anymore. So we don't require PHP 5.x support. :)

Thanks!